src/ajax.js

```diff
@@ -557,8 +557,14 @@ jQuery.extend( {
 			}
 		}
 
+		// Uppercase the type
+		s.type = s.type.toUpperCase();
+
+		// Determine if request has content
+		s.hasContent = !rnoContent.test( s.type );
+
 		// Convert data if not already a string
-		if ( s.data && s.processData && typeof s.data !== "string" ) {
+		if ( s.data && ( s.processData || !s.hasContent ) && typeof s.data !== "string" ) {
```

Comment:
in jquery ajax, data will transfered like a=1&b=2, when post ajax with contentType equal application/json, data want transfered like {"a":1, "b":2}, get ajax still use a=1&b=2

system is huge, we want process it in one place, but there is no api do this.